### PR TITLE
Add the HTML element option to columns and column block

### LINKED ISSUE #33343: "HTML Element" option for Columns and Columns blocks
Both the Columns and its inner Column block will often be used for structural sections on the page.  Users/Themers should be able to set the HTML element used.

Currently, both blocks are wrapped in a `<div>`.  However, `<main>`, `<section>`, `<aside>`, and `<article>` will make sense in various use cases.  

`<header>` and `<footer>` could potentially make sense for the outer Columns block, but I'd like to focus on getting the other options in for sure.

### PR DESCRIPTION
**Pull request by carolinan**
## Description

Adds the HTML element option to the columns and column blocks.
Closes #33343

## How has this been tested?

1. Add a columns block
2. Open the advanced section and change the HTML element
3. Save and view the front. Confirm that the selected HTML element is used.

## Types of changes

Enhancement

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by Mamaduka** [comment]
Sidenote: Now that four blocks have this option. Should we extract it as a custom component in the block-editor package?

**Comment by carolinan** [comment]
> Sidenote: Now that four blocks have this option. Should we extract it as a custom component in the block-editor package?

[> I wonder if it would also make sense to move the HTML filter to its own separate control. It is used but slightly differently for the group, template part and query blocks.](https://github.com/WordPress/gutenberg/issues/33343#issuecomment-929769608)

Possibly. I am not sure I can do that though.

**Comment by Mamaduka** [comment]
Sorry, I miss that comment. Happy to do it as a follow-up PR.

**Comment by carolinan** [comment]
I ran `npm run fixtures:regenerate test/integration/full-content/full-content.test.js`

But I then get changes like these which I don't believe are correct? I am using Windows...
```
test\integration\fixtures\blocks\core__column.json
-		"originalContent": "<div class=\"wp-block-column\">\n\t\n\t\n</div>"
+		"originalContent": "<div class=\"wp-block-column\">\r\n\t\r\n\t\r\n</div>"
```

**Comment by Mamaduka** [comment]
I just tried to regenerate fixtures locally. I get to errors related to ` core__column__deprecated*` files, and fixtures for other blocks also get updated.

Unfortunately, this is all my knowledge of fixture regeneration - run command and hope it works.

**Comment by carolinan** [comment]
yes I am seeing two errors for `core__column__deprecated*` too

**Comment by mtias** [comment]
Not to block this PR, but I'm curious if we should aim for making Column be just a variation of Group with some layout settings locked and inheriting everything else (html element, background tools, etc). cc @youknowriad

**Comment by youknowriad** [comment]
@mtias yes, this makes sense to me, the `Columns` block is nothing more than a container (group) of group blocks with a specific layout. We need to make sure the "columns" blocks is using the "flex" layout, make sure the "width" is supported for group blocks too and see how/if we can migrate.

File: packages/block-library/src/column/edit.js [hunk 1]
```diff
@@ -20,12 +21,30 @@ import {
...
+	),
+};
+
 function ColumnEdit( {
-	attributes: { verticalAlignment, width, templateLock = false },
+	attributes: {
+		verticalAlignment,
+		width,
+		templateLock = false,
+		tagName: TagName = 'div',
```
Comment (talldan) [inline packages/block-library/src/column/edit.js]: I see the attribute has a default, so I wonder whether the default value is needed here.

File: packages/block-library/src/columns/edit.js [hunk 2]
```diff
@@ -52,14 +54,33 @@ import {
...
 	setAttributes,
 	updateAlignment,
 	updateColumns,
 	clientId,
 } ) {
-	const { isStackedOnMobile, verticalAlignment } = attributes;
+	const {
+		isStackedOnMobile,
+		verticalAlignment,
+		tagName: TagName = 'div',
```
Comment (talldan) [inline packages/block-library/src/columns/edit.js]: Ditto the previous comment here.

**Review by talldan — COMMENTED** [review]
The test failure for `core__columns__deprecated.json` looks correct. The file represents block after it has been upgraded (the deprecation has already been handled), so its attributes reflect the current block.json with the new tagName attribute. To get it passing, delete the file and then run the command to regenerate the fixture.

That might also make the other test failure pass (🤞). My guess is that one is failing because it tries to serialize the block from `core__columns__deprecated.json` to HTML, but the `tagName` attribute is `undefined`, which causes the react error.

**Comment by carolinan** [comment]
I removed the unnecessary constant but I am not able to solve the problem with the fixtures.

**Pull request closed without merging**